modules/Oculus_module/include/HeadRetargeting.hpp

```diff
@@ -19,27 +19,81 @@
 // iCub-ctrl
 #include <iCub/ctrl/minJerkCtrl.h>
 
+// iDynTree
+#include <iDynTree/Core/Rotation.h>
+
+#include <RetargetingController.hpp>
+
 /**
  * Class useful to manage the head retargeting.
  */
-class HeadRetargeting
+class HeadRetargeting : public RetargetingController
 {
 private:
+    /** Minimum jerk trajectory smoother for the desired head joints */
     std::unique_ptr<iCub::ctrl::minJerkTrajGen> m_headTrajectorySmoother{nullptr};
 
-    double m_playerOrientation;
-    yarp::sig::Vector m_desiredHeadOrientation;
+
+    // In order to understand the transform defined the following frames has to be defined
+    // oculusInertial frame: it is the inertial frame of the oculus and it is placed in the
+    //                       initial position of the ovrheadset. The z axis points upward while
+    //                       x point forward
+    // teleop frame: this frame has the same origin of the oculusInertial frame but it is
+    //               rigidly attached to the user. In details the oculusInertial frame and
+    //               the teleoperation frame differs for a rotation along the Z axis.
+    // headOculus frame: frame attached to the oculus head.
+    iDynTree::Rotation m_oculusInertial_R_teleopFrame;
+    iDynTree::Rotation m_oculusInertial_R_headOculus;
+    iDynTree::Rotation m_teleopFrame_R_headOculus;
 
 public:
-    bool configure(const yarp::os::Searchable& config);
+    /**
+     * Configure the object.
+     * @param config is the reference to a resource finder object.
+     * @param name is the name of the robot.
+     * @return true in case of success and false otherwise.
+     */
+    bool configure(const yarp::os::Searchable& config, const std::string& name) override;
 
+    /**
+     * Set the player orientation
+     * @param playerOrientation is the orientation of the player in radiant (positive angle
+     * correspond to a clockwise rotation)
+     */
     void setPlayerOrientation(const double& playerOrientation);
 
-    void setDesiredHeadOrientation(const yarp::sig::Vector& desiredHeadOrientation);
+    /**
+     * Set the desired head orientation.
+     * @param oculusInertial_T_headOculus is the homogeneous transformation between the oculus
+     * inertial frame and the head oculus frame
+     */
+    void setDesiredHeadOrientation(const yarp::sig::Matrix& oculusInertial_T_headOculus);
+
+    /**
+     * Evaluate the inverse kinematics of the head
+     * @param matrix is the rotation matrix of the head with respect the root frame
+     * @param neckPitch neck pitch angle expressed in radiant
+     * @param neckRoll neck roll angle expressed in radiant
+     * @param neckYaw neck yaw angle expressed in radiant
+     */
+    static void inverseKinematics(const iDynTree::Rotation& matrix, double& neckPitch,
```

Comment:
Actually, I didn't understand. In the code the joints are named `neckPitch`, `neckRoll` and `neckYaw` while in the [iCubWiki](http://wiki.icub.org/wiki/ICub_joints) the only difference is that  `underscore_case` is used rather than the `cammelCase`.